src/Services/nssService.ts

```diff
@@ -787,6 +789,41 @@ module StreamStats.Services {
             }
             return result;
         }
+
+        // get region list form nssservices/regions
+        // referenced getFreshDeskArticles from HelpControllers
+        // keys to success: use ng.Ipromise<any> and return this.Execute 
+        public getRegionList(): ng.IPromise<any> {
+
+            var url = configuration.baseurls['NSS'] + configuration.queryparams['NSSRegions'];
+            var request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true);
+            
+            return this.Execute(request).then(
+                (response: any) => {
+                    // create array to return
+                    var regionsRaw = []; 
+
+                    // loop through response and add data to regions array
+                    response.data.forEach(function (element) {
+                        regionsRaw.push(element);
+                    });

```

Comment:
You should be able to change these lines of code to:
`var regionsRaw = response.data; `
which should be faster than a loop.